src/input-select/input-range-select.tsx

```diff
@@ -0,0 +1,311 @@
+    // =============================================================================
+    const handleSelectorClick = (
+        event: React.MouseEvent,
+        rangeType?: RangeType
+    ) => {
+        event.stopPropagation();
+        event.preventDefault();
+        if (disabled || readOnly) {
+            return;
+        }
+        if (rangeType === "from") {
+            setFocusedInput("from");
+        } else if (rangeType === "to" && selectedFromValue) {
+            setFocusedInput("to");
+        } else if (rangeType === "to" && !selectedFromValue) {
+            setFocusedInput("from");
+        } else {
+            setFocusedInput("from");
+        }
+    };
+
+    const handleListItemClick = (
+        item: T,
+        extractedValue: V,
+        rangeType?: RangeType
+    ) => {
+        rangeType === "from"
+            ? setSelectedFromValue(item)
+            : setSelectedToValue(item);
+        triggerOptionDisplayCallback(false);
+        if (selectorRef) {
+            selectorRef.current.focus();
+        }
+        if (onSelectOption) {
+            onSelectOption({ [rangeType]: item }, extractedValue);
+        }
+        if (rangeType === "from") {
+            setSelectedToValue(undefined);
+            setFocusedInput("to");
+            triggerOptionDisplayCallback(true);
+        } else {
+            setFocusedInput("none");
+        }
+    };
+
+    const handleListDismiss = (rangeType: RangeType) => {
+        setFocusedInput("none");
+        triggerOptionDisplayCallback(false);
+
+        if (selectorRef) {
+            selectorRef.current.focus();
+        }
+
+        if (!selectedFromValue || !selectedToValue) {
+            setSelectedToValue(undefined);
+            setSelectedFromValue(undefined);
+        }
+    };
+
+    const handleWrapperBlur = () => {
+        triggerOptionDisplayCallback(false);
+        setFocusedInput("none");
+        if (!selectedFromValue || !selectedToValue) {
+            setSelectedToValue(undefined);
+            setSelectedFromValue(undefined);
+        }
+    };
+
+    // =============================================================================
+    // HELPER FUNCTION
+    // =============================================================================
+    const getDisplayValue = (rangeType: RangeType): string | V => {
+        const selected =
+            rangeType === "from" ? selectedFromValue : selectedToValue;
+        if (displayValueExtractor) {
+            return displayValueExtractor(selected);
+        }
+
+        if (valueExtractor) {
+            return valueExtractor(selected);
+        }
+        return selected?.toString();
+    };
+
+    const convertValueToString = (value: V | string): string => {
+        if (typeof value === "string") {
+            return value;
+        } else {
+            return valueToStringFunction(value) || value.toString();
+        }
+    };
+
+    const truncateValue = (type: RangeType, value: string | V) => {
```

Comment:
Observed that the truncation did not get updated when the widths changed. E.g. if it was not truncated at the beginning and you go into the compressed width, it doesn't get truncated. It only truncates if you refresh the page in the compressed width. And when the width increases, the truncation still remains. 

Reason being there has to be a re-render (via state change) for the truncation to be updated. We can ignore this for now and accept it as a known bug. Perhaps we'll have to spin up a shared component that can truncate responsively or something.

Reply:
Yes, noticed the same thing. agree, also agree to take it up as separate bug card.